build/generate-cli-docs/generate.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cobra/doc"
+
+	"github.com/open-policy-agent/opa/cmd"
+)
+
+var fileHeader = `---
+title: CLI
+kind: documentation
+weight: 90
+restrictedtoc: true
+---
+
+The OPA executable provides the following commands.
+
+`
+
+func main() {
+	if len(os.Args) != 2 {
+		log.Fatal("Required argument: cli docs output directory")
+	}
+	out := os.Args[1]
+
+	command := cmd.RootCommand
+	command.Use = "opa [command]"
+	command.DisableAutoGenTag = true
+
+	dir, err := ioutil.TempDir("", "opa")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	err = doc.GenMarkdownTree(command, dir)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	builder := strings.Builder{}
+
+	last := len(files) - 1
+	for i, file := range files {
+		path := filepath.Join(dir, file.Name())
+		document, err := fixupSection(path)
+		if err != nil {
+			log.Fatal(err)
+		}
+		builder.WriteString(document)
+		if i != last {
+			builder.WriteString("____\n\n")
+		}
+	}
+
+	heading := regexp.MustCompile(`^[\\-]+$`)
+	var document []string
+	removed := 0
+
+	// The document may contain "----" for headings, which will be converted to h1
+	// elements in markdown. This is undesirable, so let's remove them and prepend
+	// the line before that with ### to instead create a h3
+	for line, str := range strings.Split(builder.String(), "\n") {
+		if heading.Match([]byte(str)) {
+			document[line-1-removed] = fmt.Sprintf("### %s\n", document[line-1-removed])
+			removed++
+			continue
+		}
+		document = append(document, fmt.Sprintf("%s\n", str))
+	}
+
+	withHeader := fmt.Sprintf("%s%s", fileHeader, strings.Join(document, ""))
+	err = ioutil.WriteFile(filepath.Join(out, "cli.md"), []byte(withHeader), 0755)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func fixupSection(path string) (string, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+	builder := strings.Builder{}
+
+	for scanner.Scan() {
+		// Remove "See also" section
+		if strings.Contains(scanner.Text(), "### SEE ALSO") {
+			break
+		}
+		builder.WriteString(scanner.Text())
+		builder.WriteString("\n")
+	}
+	if err := scanner.Err(); err != nil {
+		return "", err
+	}
+
+	return builder.String(), nil
```

Comment:
[nit]
```suggestion
	return builder.String(), scanner.Err()
```

But this is code golf, please ignore 😄

Reply:
Heh, it's nice :) I'll have it changed.